src/Kafka/ManageKafkaPermissions/components/SelectAccount.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Account, PrincipalType } from "../types";
+import {
+  Divider,
+  Select,
+  SelectGroup,
+  SelectOption,
+  SelectVariant,
+  ValidatedOptions,
+  SelectProps,
+  SelectOptionObject,
+} from "@patternfly/react-core";
+
+export type Validated<T> = {
+  value: T;
+  validated?: ValidatedOptions;
+  errorMessage?: string;
+};
+
+export type SelectAccountProps = {
+  id: string | undefined | SelectOptionObject;
+  onChangeAccount: React.Dispatch<
+    React.SetStateAction<string | undefined | SelectOptionObject>
+  >;
+  accounts: Account[];
+  onEscapeModal: (closes: boolean) => void;
+  validated: ValidatedOptions;
+  onChangeValidation: (value: ValidatedOptions) => void;
```

Comment:
Not sure why the UI component should tell the parent if the selected value is valid or not. It should be the parent that knows if the value is valid, and the validation state is indeed being passed.